### PR TITLE
Add a blog post about packaging a web site and nginx

### LINKED ISSUE #2878: how to use nginx with a web application
Users are currently confused about how to leverage `core/nginx` with their web applications. We need to have a step by step guide explaining how to use `core/nginx` in a best practices way. 

Acceptance criteria: 
- Step by step blog post
- hand off blog post to @davidwrede to turn into "Getting Started" docs

### PR DESCRIPTION
**Pull request by cnunciato**
Signed-off-by: Christian Nunciato <cnunciato@chef.io>

Resolves #2878.

### CODE REVIEW

**Comment by cnunciato** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/263604411).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/263604411).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by cnunciato** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/263635099).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/263635099).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by cnunciato** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/263828493).

**Comment by thesentinels** [comment]
:broken_heart: Travis CI reports [this PR failed to deploy](https://travis-ci.org/habitat-sh/habitat/builds/263828493).

The next step is to examine the [job](https://travis-ci.org/habitat-sh/habitat/builds/263828493) and figure out why. If it is transient, you can try re-triggering the Travis CI Job - if it passes, this PR will be automatically merged. If it is not transient, you should fix the issue and update this pull request, and issue `approve` again. If you believe it will never pass, and you are feeling :godmode:, you can issue a `force` to merge this PR anyway.

**Comment by cnunciato** [comment]
That failure was actually a transient failure to download `rust-docs`; the deploy succeeded and is viewable here: http://www.acceptance.habitat.sh/blog/2017/08/packaging-a-web-site-with-nginx-and-habitat/

**Comment by mwrock** [comment]
I liked the jokes in the 4 step process at the beginning that was there before!

File: www/source/blog/2017-08-11-Packaging-a-Web-Site-with-Nginx-and-Habitat.html.md [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+    ↑ Applying to peer 127.0.0.1:9638
+    ★ Applied configuration
+
+Now go ahead and give that browser a reload, and you should finally see what you've been waiting for:
+
+![](media/2017-08-11-Hab-and-Nginx/hello-hab.png)
+
+## Summing Up
+
+That was a whole bunch of words to explain what was really just a few lines of code (which, by the way, [is up on GitHub](https://github.com/cnunciato/hello-hab) if you'd like to see at everything in context), and we've covered a lot. We've learned:
```
Comment (mwrock) [inline www/source/blog/2017-08-11-Packaging-a-Web-Site-with-Nginx-and-Habitat.html.md]: `like to see at everything in context` => `like to see everything in context`

**Comment by mwrock** [comment]
ok there were a couple comic gems in there so I'm satisfied. Overall I really like the post. Its an interesting pattern: separately abstracting the server from the site. What I like about that is that you can easily reuse the hooks and config in the service for several different sites. Its biggest draw back in my mind is that an update strategy on the server will not pick up website updates. Until we figure this out with "composite" plans, I suppose you would need to fuse the server and website plan into one in order to leverage an update summary.

**Comment by tashimi** [comment]
We should probably add Matt's comments about the composite plan that's in development & link to the roadmap & github project tracker for interested people to see progress on that.

**Comment by cnunciato** [comment]
Ha! Thanks for reading through it.

Yes, this is true, and I definitely intended to have people run `hab config apply` to update the server to use the packaged site -- and to run it again to pick up changes to the site. But you're right, that is a bit of a pain.

One thing I could do is change the post to make the "static" site runnable as a service (e.g., as a Node script), or as you suggest (and which I've seen others do), just bake the two packages together. But as a developer, I don't love either of those options; I'd rather ship them separately and have that be something we try to make a bit nicer for packages that aren't services, either with composite plans or something else.

**Comment by mwrock** [comment]
Another option one would have here is leveraging scaffolding. The scaffolding would have the reusable service code that pulls in node and sets up the hooks and the site would consume that scaffolding. However, I think including content that walks users through creating a scaffolding plan would be beyond the scope of this post.

**Comment by cnunciato** [comment]
Yeah, I'd love to pull in scaffolding. I was thinking of that as a follow-up post, maybe to add demonstrate adding an API for the UI to consume.

**Comment by mwrock** [comment]
Yeah  follow up post would be perfect! You could just mentioon the "caveat" here with a "stay tuned" for the scaffolding.

**Comment by cnunciato** [comment]
@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/264757431).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/264757431).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by cnunciato** [comment]
All right, I think I'm done with this. Please review and if it looks good, merge away, so I don't have to keep changing the filename and frontmatter. 😄  Thanks for all the eyeballs!

**Comment by cnunciato** [comment]
http://www.acceptance.habitat.sh/blog/2017/08/packaging-a-website-with-nginx-and-habitat/

**Comment by christophermaier** [comment]
As we're currently conceiving the "composite packages" feature, I don't quite think that it would address the use case presented here, since the static site isn't a live service.

However, I think that you could probably get the updating that you want by having your static site artifact be a dependency of your `hello-hab-server`... that way, when you pushed new builds of your static content, it'd trigger a re-build and redeploy of your `hello-hab-server` service. We may find better ways to address that use case in the future, but that seems like a viable way forward now.

**Comment by mwrock** [comment]
That works but then it seems like one is back to square one where the server is coupled to the site because its directly taking a dependency on the site instead of being a reusable server component for multiple sites.

I really like the scaffolding approach here because:
1. you get the server reuse in a scaffold but the scaffold dependency is declared in the site.
2. If explored in a future post it opens up a topic that has not been discussed much at all: creating your own scaffolds for your internal or customized apps that might not be appropriate for `core-plans`.

**Comment by cnunciato** [comment]
All right, updated with a note about scaffolding support.

@thesentinels acceptance

**Comment by thesentinels** [comment]
:metal: I am sending your branch against master to our acceptance environment.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started the deployment of acceptance](https://travis-ci.org/habitat-sh/habitat/builds/264916661).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports acceptance has been deployed](https://travis-ci.org/habitat-sh/habitat/builds/264916661).

I love new and shiny websites!

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Comment by mwrock** [comment]
Looks great!

@thesentinels approve

**Comment by thesentinels** [comment]
:metal: I am testing your branch against master before merging it. We do this to ensure that the master branch is never failing tests.

**Comment by thesentinels** [comment]
:neckbeard: Travis CI has [started testing this PR](https://travis-ci.org/habitat-sh/habitat/builds/264931404).

**Comment by thesentinels** [comment]
:sparkling_heart: Travis CI [reports this PR passed](https://travis-ci.org/habitat-sh/habitat/builds/264931404).

It always makes me feel nice when humans approve of one anothers work. I'm merging this PR now.

I just want you and the contributor to answer me one question:

gif-keyboard-3280869874741411265

**Pull request merged**